src/everest/suite.py

```diff
@@ -389,38 +392,68 @@ def start_optimization(self):
         """
         assert self._monitor_thread is None
 
-        # Initialize the Everest simulator:
-        simulator = Simulator(self.config, callback=self._simulation_callback)
-
-        # Initialize the ropt optimizer:
-        optimizer = self._configure_optimizer(simulator)
-
-        # Before each batch evaluation we check if we should abort:
-        optimizer.add_observer(
-            EventType.START_EVALUATION,
-            partial(self._ropt_callback, optimizer=optimizer, simulator=simulator),
+        config_dict = everest_to_ert_config(
+            self.config, site_config=ErtConfig.read_site_config()
         )
-
-        # The SqliteStorage object is used to store optimization results from
-        # Seba in an sqlite database. It reacts directly to events emitted by
-        # Seba and is not called by Everest directly. The stored results are
-        # accessed by Everest via separate SebaSnapshot objects.
-        # This mechanism is outdated and not supported by the ropt package. It
-        # is retained for now via the seba_sqlite package.
-        seba_storage = SqliteStorage(optimizer, self.config.optimization_output_dir)
-
-        # Run the optimization:
-        exit_code = optimizer.run().exit_code
-
-        # Extract the best result from the storage.
-        self._result = seba_storage.get_optimal_result()
-
-        if self._monitor_thread is not None:
-            self._monitor_thread.stop()
-            self._monitor_thread.join()
-            self._monitor_thread = None
-
-        return "max_batch_num_reached" if self._max_batch_num_reached else exit_code
+        ert_config = ErtConfig.with_plugins().from_dict(config_dict=config_dict)
+
+        # Inject ExtParam nodes. This is needed because EXT_PARAM is not an ERT
+        # configuration key, but only a placeholder for the control definitions.
+        ens_config = ert_config.ensemble_config
+        for control_name, variables in config_dict["EXT_PARAM"].items():
+            ens_config.addNode(
+                ExtParamConfig(
+                    name=control_name,
+                    input_keys=variables,
+                    output_file=control_name + ".json",
+                )
+            )
+
+        with open_storage(ert_config.ens_path, mode="w") as storage:
+            # Initialize the Everest simulator:
+            experiment = storage.create_experiment(
+                name=f"EnOpt@{datetime.datetime.now().strftime('%Y-%m-%d@%H:%M:%S')}",
+                parameters=ert_config.ensemble_config.parameter_configuration,
+                responses=ert_config.ensemble_config.response_configuration,
+            )
+
+            simulator = Simulator(
+                self.config,
+                ert_config,
+                storage,
+                experiment,
+                callback=self._simulation_callback,
+            )
```

Comment:
I don't really like this. This has gotten a lot more complicated now. Before this was a single call to instantiate Simulator, now it exposes a lot of the machinery to run the ensemble. I understand that in the future we may want access to the storage (and experiment?) outside of the simulator object. Can we encapsulate this somehow?

Reply:
Hmmm..  Some encapsulation could be done, but I think we do need to be able to:

1. Run an Everest optimization within the context of one write-access storage
2. Initialize the storage outside of the async callbacks etc
3. For future (3) we will be able to link up ROPT callbacks with an existing experiment / ensemble which it can write optimization results into. Currently ROPT initialization is done in `suite.py`, if we were to move all of this into `Simulator`, we would also have to pack the ROPT stuff in there to achieve a "correct" order of initialization so we can link things up.

Currently, open_storage is called after some async call from ROPT to a callback in ERT, and the experiment was created then. If we are to have one experiment per optimization run, and one ensemble per batch (for which we store some optimization results), we need to access the experiment (which we will write to) when we set up the ROPT observers. These observers need to be added before calling `simulator.start()` currently, and we can not add listeners afterwards. So I would say it makes sense to have the entire optimization within the context of one open storage. 

There are also some internal things in storage that makes it a bit harder to keep it encapsulated (these may be revised away on the storage side):

1. We can only have one storage with write-access open at a time. If each batch is to write its own results, we would need multiple write-storages, or a method that writes optimization results without "write access" (easy to achieve but not that clean).
5. Experiments without any ensembles will not be loaded from the file system, this makes it impossible to initialize an empty experiment first, and then add ensembles to it by re-opening a write storage (the re-opened storage will not load the experiment that was created in the first place, we could create a dummy empty ensemble just to make it load, or add something to storage to make it load "empty" experiments.
6. The experiment needs a response / parameter config, which requires the `ExtParamConfig`s to be added to the ERT config prior to initializing the experiment.

Reply:
Hmmm.. We could hold the experiment in the simulator and have the same functionality, will set it up.